packages/systemd/0006-udev-requeue-event-when-the-corresponding-block-devi.patch

```diff
@@ -0,0 +1,286 @@
+  e.g. new devlinks are not created, or old devlinks are not removed.
+
+This commit makes events are requeued with a tiny delay when the corresponding
+block devices are locked by other processes. With this way, the triggered
+uevent may be delayed but is always processed by udevd. Hence, the above
+issues can be solved. Also, it is not necessary to watch a block device
+unconditionally when it is already locked. Hence, the logic is dropped.
+---
+ src/udev/udevd.c | 154 +++++++++++++++++++++++++++++------------------
+ 1 file changed, 97 insertions(+), 57 deletions(-)
+
+diff --git a/src/udev/udevd.c b/src/udev/udevd.c
+index abf50b6a71..01d782421e 100644
+--- a/src/udev/udevd.c
++++ b/src/udev/udevd.c
+@@ -68,6 +68,8 @@
+ #include "version.h"
+ 
+ #define WORKER_NUM_MAX 2048U
++#define EVENT_RETRY_INTERVAL_USEC (200 * USEC_PER_MSEC)
++#define EVENT_RETRY_TIMEOUT_USEC  (3 * USEC_PER_MINUTE)
+ 
+ static bool arg_debug = false;
+ static int arg_daemonize = false;
+@@ -126,6 +128,8 @@ typedef struct Event {
+         sd_device_action_t action;
+         uint64_t seqnum;
+         uint64_t blocker_seqnum;
++        usec_t retry_again_next_usec;
++        usec_t retry_again_timeout_usec;
+ 
+         sd_event_source *timeout_warning_event;
+         sd_event_source *timeout_event;
+@@ -150,8 +154,13 @@ typedef struct Worker {
+ } Worker;
+ 
+ /* passed from worker to main process */
+-typedef struct WorkerMessage {
+-} WorkerMessage;
++typedef enum EventResult {
++        EVENT_RESULT_SUCCESS,
++        EVENT_RESULT_FAILED,
++        EVENT_RESULT_TRY_AGAIN, /* when the block device is locked by another process. */
++        _EVENT_RESULT_MAX,
++        _EVENT_RESULT_INVALID = -EINVAL,
++} EventResult;
+ 
+ static Event *event_free(Event *event) {
+         if (!event)
+@@ -365,10 +374,11 @@ static void device_broadcast(sd_device_monitor *monitor, sd_device *dev) {
+                                          "Failed to broadcast event to libudev listeners, ignoring: %m");
+ }
+ 
+-static int worker_send_message(int fd) {
+-        WorkerMessage message = {};
++static int worker_send_result(Manager *manager, EventResult result) {
++        assert(manager);
++        assert(manager->worker_watch[WRITE_END] >= 0);
+ 
+-        return loop_write(fd, &message, sizeof(message), false);
++        return loop_write(manager->worker_watch[WRITE_END], &result, sizeof(result), false);
+ }
+ 
+ static int worker_lock_block_device(sd_device *dev, int *ret_fd) {
+@@ -493,44 +503,12 @@ static int worker_process_device(Manager *manager, sd_device *dev) {
+         if (!udev_event)
+                 return -ENOMEM;
+ 
++        /* If this is a block device and the device is locked currently via the BSD advisory locks,
++         * someone else is using it exclusively. We don't run our udev rules now to not interfere.
++         * Instead of processing the event, we requeue the event and will try again after a delay.
```

Comment:
Need to confirm via testing that this doesn't interfere with the previous fix to the fallback logic, since that relies on continued udev processing to generate the special block-device-typed UUID symlinks.

My recollection is that all the symlinks are created at the end, so we shouldn't see the case where the fallback unit starts execution as soon as the partuuid link exists, but before the typed link exists.